### PR TITLE
Dialog: Add autoheight class if Height set to 'auto'

### LINKED ISSUE #915: Setting height on Dialog gets overriden.
# 🐛 Bug Report

When setting the height on a dialog and having the alignment set to Center the height property is overridden with height:auto.

## 💻 Repro or Code Sample

```
          IDialogReference dialog = await DialogService.ShowDialogAsync(typeof(SearchDialog<T>), request, new DialogParameters()
          {
              Title = ModalTitle ?? "Search",
              Alignment = HorizontalAlignment.Center,
              Width = "90vw",
              Height = "90vh"
          });
```

## 🤔 Expected Behavior

Should be set to the height requested.

## 😯 Current Behavior

Is set to auto.

image

### PR DESCRIPTION
**Pull request by vnbaaij**
Resolve #915 by adding an autoheight class when Height set to auto. Also added a test for this.

### CODE REVIEW

File: src/Core/Components/Dialog/FluentDialog.razor.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public partial class FluentDialog : FluentComponentBase
         .AddClass("right", () => _parameters.DialogType == DialogType.Panel && _parameters.Alignment == HorizontalAlignment.Right)
         .AddClass("left", () => _parameters.DialogType == DialogType.Panel && _parameters.Alignment == HorizontalAlignment.Left)
         .AddClass("prevent-scroll", () => Instance is null ? (PreventScroll && !Hidden): _parameters.PreventScroll)
+        .AddClass("autoheight", () => !string.IsNullOrWhiteSpace(_parameters.Height) && _parameters.Height == "auto")
```
Comment (dvoituron) [inline src/Core/Components/Dialog/FluentDialog.razor.cs:27]: Probably `() => _parameters.Height == "auto"` is enough. Why to add an extra test on "Not Empty"?

File: src/Core/Components/Dialog/FluentDialog.razor.css [hunk 2]
```diff
@@ -3,12 +3,13 @@
     position: relative;
     padding: 0px var(--dialog-padding);
     width: calc(var(--dialog-width) - 2 * var(--dialog-padding));
-    height: calc(var(--dialog-height) - 2 * var(--dialog-padding));
-    max-width: 80%;
 }
```
Comment (dvoituron) [inline src/Core/Components/Dialog/FluentDialog.razor.css:6]: Why did you remove `max-width` ?

File: src/Core/Components/Dialog/FluentDialog.razor.css [hunk 3]
```diff
@@ -3,12 +3,13 @@
...
 
-fluent-dialog:has(div[class~="fluent-dialog-body"]):not(:is(fluent-dialog[class~="right"], fluent-dialog[class~="left"]) )::part(control) {
+fluent-dialog.autoheight::part(control) {
```
Comment (dvoituron) [inline src/Core/Components/Dialog/FluentDialog.razor.css:8]: When `Height` is a specified value. How the height is set, if you removed `height: calc(var(--dialog-height) - 2 * var(--dialog-padding));`?

Is it possible to keep the height in `fluent-dialog::part(control)` and to override with this style and an `!important` flag?

File: tests/Core/Dialog/FluentDialogServiceTests.razor [hunk 4]
```diff
@@ -23,4 +23,26 @@
         // Assert
         cut.Verify();
     }
+
+    [Fact]
```
Comment (dvoituron) [inline tests/Core/Dialog/FluentDialogServiceTests.razor:27]: Probably interesting to add a second test with a specific Height (ex: `Height="240px").

File: src/Core/Components/Dialog/FluentDialog.razor.css (hunk 2 above)
Comment (vnbaaij) [reply src/Core/Components/Dialog/FluentDialog.razor.css:6]: Because on mobile I want a dialog or panel to take the whole width of the screen. Needs more changes probably

File: src/Core/Components/Dialog/FluentDialog.razor.css (hunk 3 above)
Comment (vnbaaij) [reply src/Core/Components/Dialog/FluentDialog.razor.css:8]: Height is then set as `--dialog-height` in the style of the fluent-dialog. There is actually no need to set the heigth in the  part(control) again

File: src/Core/Components/Dialog/FluentDialog.razor.cs (hunk 1 above)
Comment (vnbaaij) [reply src/Core/Components/Dialog/FluentDialog.razor.cs:27]: You are right. Will change that

File: tests/Core/Dialog/FluentDialogServiceTests.razor (hunk 4 above)
Comment (vnbaaij) [reply tests/Core/Dialog/FluentDialogServiceTests.razor:27]: Yes, will do

Comment (vnbaaij) [reply tests/Core/Dialog/FluentDialogServiceTests.razor:27]: The test above that one already has a height set

File: src/Core/Components/Dialog/FluentDialog.razor.css (hunk 2 above)
Comment (dvoituron) [reply src/Core/Components/Dialog/FluentDialog.razor.css:6]: In this case, we need to add a `@media` style to remove the `max-width` on mobile.
And I'm not even sure it's a good idea to accept 100% on mobile. Because a DialogBox should look like a window, even on mobile (in my opinion).

File: src/Core/Components/Dialog/FluentDialog.razor.css (hunk 3 above)
Comment (dvoituron) [reply src/Core/Components/Dialog/FluentDialog.razor.css:8]: Including the `2 * var(--dialog-padding)`? So, why to keep `width`?

**Review by dvoituron — APPROVED** [review]
